src/pages/GivePage/queries.ts

```diff
@@ -122,6 +122,141 @@ export type CircleAllocationText = Awaited<
 >;
 export const QUERY_KEY_CIRCLE_ALLOCATION_TEXT = 'getCircleAllocationText';
 
+export const getPendingGiftsFrom = async (
+  selectedCircleId: number,
+  address: string
+) => {
+  const data = await client.query(
+    {
+      pending_token_gifts: [
+        {
+          where: {
+            sender_address: {
+              _ilike: address,
+            },
+            circle_id: {
+              _eq: selectedCircleId,
+            },
+          },
+        },
+        {
+          tokens: true,
+          recipient_id: true,
+          gift_private: {
+            note: true,
+          },
+        },
+      ],
+    },
+    {
+      operationName: 'pendingGiftsFrom',
+    }
+  );
+
+  type GiftWithNote = Omit<
+    typeof data.pending_token_gifts[number],
+    'gift_private' | 'tokens'
+  > & { tokens?: number; note?: string };
+
+  return data.pending_token_gifts.map(g => {
+    const gm = g;
+    const note = g.gift_private?.note;
+    // have to delete this field because spread operator includes it elsewhere and it causes validation issues -g
+    delete gm['gift_private'];
+    const gwn: GiftWithNote = {
+      ...gm,
+      note,
+    };
+    return gwn;
+  });
+};
+
+export type PendingGift = Awaited<
+  ReturnType<typeof getPendingGiftsFrom>
+>[number];
+
+export const getTeammates = async (
+  selectedCircleId: number,
+  address: string
+) => {
+  const data = await client.query(
+    {
+      users: [
+        {
+          where: {
+            circle_id: { _eq: selectedCircleId },
+            address: { _nilike: address },
+          },
+        },
+        {
+          id: true,
+          circle_id: true,
+          name: true,
+          bio: true,
+          address: true,
+          non_receiver: true,
+          fixed_non_receiver: true,
+          created_at: true,
+          role: true,
+          profile: {
+            avatar: true,
+            address: true,
+            id: true,
+            name: true,
+          },
+        },
+      ],
+      teammates: [
+        {
+          where: {
+            user: {
+              circle_id: { _eq: selectedCircleId },
+              address: { _ilike: address },
+            },
+          },
+        },
+        {
+          team_mate_id: true,
+          teammate: {
+            // id: true,
```

Comment:
why commented? remove if not needed?

Reply:
this whole query can be deleted , nice catch